From 84f144bedd81ad154a26c76fb322903c25374d20 Mon Sep 17 00:00:00 2001
From: Varun Reddy Yeturu <varunreddy.yeturu@codeaurora.org>
Date: Wed, 27 Sep 2017 21:17:54 -0700
Subject: qcacld-3.0: Check for the max number of P2P NOA descriptors

Check for the maximum number of P2P NOA descriptors in
wma_send_bcn_buf_ll.

Change-Id: If7e5b3c53309412dc7d3cd748c2f5581898fbbfe
CRs-Fixed: 2114323
---
 core/wma/src/wma_mgmt.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/wma/src/wma_mgmt.c b/core/wma/src/wma_mgmt.c
index a42417e..383859d 100644
--- a/core/wma/src/wma_mgmt.c
+++ b/core/wma/src/wma_mgmt.c
@@ -103,6 +103,12 @@ static void wma_send_bcn_buf_ll(tp_wma_handle wma,
 		WMA_LOGE("%s: Invalid beacon buffer", __func__);
 		return;
 	}
+	if (WMI_UNIFIED_NOA_ATTR_NUM_DESC_GET(p2p_noa_info) >
+			WMI_P2P_MAX_NOA_DESCRIPTORS) {
+		WMA_LOGE("%s: Too many descriptors %d", __func__,
+			WMI_UNIFIED_NOA_ATTR_NUM_DESC_GET(p2p_noa_info));
+		return;
+	}
 
 	qdf_spin_lock_bh(&bcn->lock);
 
-- 
cgit v1.1

